src/SubdomainInput/SubdomainInput.jsx

```diff
@@ -127,6 +127,18 @@ export const SubdomainInput = forwardRef(
     }, [value, ref])
 
     const handleInput = event => {
+      const { innerText } = event.target
+      const el = (ref || inputRef).current
+
+      el.innerText = innerText.trim()
+      replaceCaret(el)
+
+      if (type === 'number' && isNaN(innerText)) {
+        el.innerText = innerText.substring(0, innerText.length - 1)
+        replaceCaret(el)
+        return
```

Comment:
this does not account for pasting multiple characters at once

Reply:
f79c19677e444acf0ec2fe4920f8d4e780eff920